src/jvm/main/org/jetbrains/kotlinx/lincheck/runner/ParallelThreadsRunner.kt

```diff
@@ -262,9 +261,26 @@ internal open class ParallelThreadsRunner(
             }
             val afterPostStateRepresentation = constructStateRepresentation()
             // Execute validation functions
-            executeValidationFunctions(testInstance, validationFunctions) { functionName, exception ->
-                return ValidationFailureInvocationResult(scenario, functionName, exception)
+            if (validationFunctions.isNotEmpty()) {
```

Comment:
Please use the same scheme as for init/post parts for the validation functions -- use the bytecode generation